sway/input/seat.c

```diff
@@ -806,7 +806,6 @@ static void seat_configure_keyboard(struct sway_seat *seat,
 	// the current focused surface to avoid breaking input grabs.
 	struct wlr_surface *surface = seat->wlr_seat->keyboard_state.focused_surface;
 	if (surface) {
-		wlr_seat_keyboard_notify_clear_focus(seat->wlr_seat);
```

Comment:
I've been using this for 4 hours with fcitx5 and haven't noticed any issues, also river also cargoculted sway logic and made the same change https://codeberg.org/river/river/commit/6263699ca8fd04f9119d8b5f163f6f1ba372975f

Reply:
If we drop the `wlr_seat_keyboard_notify_clear_focus()` call, we should also drop the `seat_keyboard_notify_enter()` call I think?

I believe the issue is that clients are confused when updating the keymap while some keys are pressed? So this forces clients to re-process the keys in their enter event handler?

Reply:
My understanding of the issue is that fcitx5 started destroying virtual keyboard when not in focus, and creating new ones when in focus. Previously I think it reused the same virtual device.

This caused sway and fcitx5 to race each other to the bottom, because fcitx5 would create a new device, then sway would send a leave/enter, causing fcitx5 to destroy and create a device again.

Removing `seat_keyboard_notify_enter` would break input grab I think, and it doesn't really make sense in my head.

Reply:
A keyboard gets configured if it is first plugged in or something about the configuration changes. If it's first plugged in, we want `seat_keyboard_notify_enter` because it means that clients will get notified immediately. Otherwise, it would be a no-op. This looks fine.